src/arrayMethodSort.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const copmareFunc = typeof compareFunction === 'function'
+      ? compareFunction
+      : (a, b) => {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA === strB) {
+          return 0;
+        } else if (strA < strB) {
+          return -1;
+        } else {
+          return 1;
+        }
+      };
```

Comment:
its better to create separate function for this, and also you shouldnt write nested if else statement
```suggestion
      : (a, b) => {
        const strA = String(a);
        const strB = String(b);

        if (strA === strB) {
          return 0;
        }
        
        if (strA < strB) {
          return -1;
        }
        
        return 1;
      };
```